src/test/java/com/wikia/webdriver/common/logging/LOG.java

```diff
@@ -0,0 +1,327 @@
+  private static ArrayList<Boolean> logsResults = new ArrayList<>();
+  private static boolean testStarted = false;
+  private static boolean allowLongDesc = false;
+
+  private static void appendToReport(String command, String description, boolean success,
+                                     boolean makeScreenshot) {
+    Type classType = success ? Type.SUCCESS : Type.ERROR;
+    appendToReport(command, description, classType, makeScreenshot);
+  }
+
+  private static void appendToReport(String command, String description, Type type,
+                                     boolean makeScreenshot) {
+    logCounter++;
+    logsResults.add(type.isError());
+    String className = type.getType();
+    StringBuilder builder = new StringBuilder();
+
+    String transformDesc = description;
+    if (new UrlValidator().isValid(description)) {
+      transformDesc = "<a href='" + description + "'>" + description;
+    }
+    boolean isDescriptionTooLong = (transformDesc.length() > MAX_CONTENT_LENGTH) && !allowLongDesc;
+
+    if (isDescriptionTooLong) {
+      builder.append("<tr class=\"accordion-toggle " + className
+                     + "\" data-toggle=\"collapse\" data-target=\"#log" + logCounter
+                     + "\" class=\"accordion-toggle\"><td><div class=\"" + type.getIconClass()
+                     + "\"></div>" + command
+                     + "</td><td> CLICK TO SEE MORE </td>");
+    } else {
+      builder.append("<tr class=\"accordion-toggle " + className
+                     + "\" data-toggle=\"collapse\" data-target=\"#log" + logCounter
+                     + "\" class=\"accordion-toggle\"><td><div class=\"" + type.getIconClass()
+                     + "\"></div>" + command + "</td><td>"
+                     + transformDesc
+                     + "</td>");
+    }
+    if (makeScreenshot && TestContext.getWebDriver() != null) {
+      new Shooter().savePageScreenshot(SCREEN_PATH + logCounter, TestContext.getWebDriver());
+      CommonUtils.appendTextToFile(SCREEN_PATH + logCounter + ".html", getPageSource());
+      builder.append("<td><a href='screenshots/screenshot" + logCounter
+                     + ".png'>Screenshot</a><br/><a href='screenshots/screenshot" + logCounter
+                     + ".html'>HTML Source</a></td></tr>");
+    } else {
+      builder.append("<td></td></tr>");
+    }
+    if (isDescriptionTooLong) {
+      builder
+          .append("<tr class=\""
+                  + className
+                  + "\"><td colspan=\"3\" class=\"hiddenRow\">"
+                  + "<div class=\"accordian-body collapse\" id=\"log"
+                  + logCounter + "\">" + description + "</div></td><tr>");
+    }
+    allowLongDesc = false;
+    CommonUtils.  appendTextToFile(LOG_PATH, builder.toString());
+  }
+
+  public static void stopLogging() {
+
+    if (TestContext.getWebDriver() != null){
+      try {
+        TestContext.getWebDriver().quit();
+      } catch (Error e) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Catch Exception instead of Error. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)